File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.js
function solution(arr) {
    let x = 0;
    
    while(true) {
        const arr_prev = [...arr]
        
        for (let i = 0; i < arr.length; i++) {
            const num = arr[i];

            if (num >= 50 && num % 2 == 0) {
                arr[i] = num / 2;
            } else if (num < 50 && num % 2 != 0) {
                arr[i] = num * 2 + 1;
            }
        }
        
        const isEqual = arr.length === arr_prev.length && 
                         arr.every((val, i) => val === arr_prev[i]);
        
        if (isEqual) return x;
        
        x += 1;
    }
}

File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.py
def solution(arr):
    x = 0
    
    while True:
        arr_prev = arr[:]
        
        for i in range(len(arr)):
            num = arr[i]
            if num >= 50 and num % 2 == 0:
                arr[i] = num // 2
            elif num < 50 and num % 2 != 0:
                arr[i] = num * 2 + 1
        
        if arr == arr_prev:
            return x
        
        x += 1